docs/advanced/compiling-wled.md

```diff
@@ -84,61 +83,53 @@ The .bin file is located in the subfolder `/build_output/firmware` in your WLED
-[ESPAsyncUDP](https://github.com/me-no-dev/ESPAsyncUDP) | ESP8266 only
-[AsyncTCP for ESP32](https://github.com/me-no-dev/AsyncTCP) | ESP32 only
-[LITTLEFS_esp32](https://github.com/lorol/LITTLEFS)* | ESP32 only
+Install `git` if it isn't already: `sudo apt install git`
 
-\* Please see [the installation guide](https://github.com/lorol/LITTLEFS#installation). You might need to enable a define in the library code.
+Install the Node.js package manager: `sudo apt install npm`
 
-All other dependencies are included with WLED for convenience.
+Run `node --version` and make sure you have at least version 20. If your Linux distribution uses an older version, install Node.js directly from [Node.js](https://nodejs.org/en/download)
 
-- Now compile and flash the software! Make sure you erase everything when you flash! (If your board config does not provide this option, you can `Sketch -> Export compiled Binary` and upload with [any ESP flashing tool](/basics/install-binary).)
+Install PlatformIO:
 
-### Compilation settings (Arduino IDE)
+!!! warning
+    Do NOT install PlatformIO with `apt`
 
-ESP8266:
+```
+curl -fsSL -o get-platformio.py https://raw.githubusercontent.com/platformio/platformio-core-installer/master/get-platformio.py
+python3 get-platformio.py # if this fails because of missing packages, install the packages it complains about, then retry
+PATH=$PATH:$HOME/.platformio/penv/bin/
+```
 
-- Arduino Core v2.7.4
-- Board: NodeMCU 1.0 (ESP-12E module) (or select your ESP board)
-- CPU frequency: 80 MHz
-- Flash size : 4MB (1MB SPIFFS)
-- LwIP variant: v1.4 Higher Bandwidth (try 2 if you experience issues)
-- Upload speed: Any, 921600 recommended
+Clone the WLED Git repository (or a different fork)
+```
+git clone https://github.com/wled/WLED.git
+cd WLED
+```
+Optional: check out the branch or tag you want to build, e.g. `git checkout v0.15.0`
 
-ESP8266-07 (External Antenna):
+Run `npm ci` to do a clean install of all the required dependencies
 
-- Variants have 512kB or 1MB flash
-- Be sure to use DOUT mode when flashing
-- Flash Size 1MB (128k SPIFFS)
-- 512kB variant no longer compatible
+#### Compiling
+
+To build all environments, simply run `pio run`. More likely though you'll want to only build an image for the specific device you're using, which you can do with something like `pio run -e esp8266_2m_160`.
+To see a list of available environments, run
+```
+pio project config --json-output | jq -cr '.[0][1][0][1]'
+```
 
-ESP-07s (External Antenna):
+#### Flashing the compiled binary
 
-- Variant has 4MB flash
-- Settings as for NodeMCU or Wemos
+The .bin and .bin.gz files are located in the subfolder `/build_output/release` in your WLED folder.
 
-ESP32:
+If you have an existing WLED install, you can flash the image using the Config -> Security & Updates -> Manual OTA Update option in the web interface. Otherwise, follow the instructions [here](/basics/install-binary/#flashing-method-2-esptool).
```

Comment:
just curious - if you build your firmware within VSCode, why would a user want to manually upload with esptool, and not use the "build and upload" button? 

see also https://github.com/wled/WLED-Docs/issues/161#issuecomment-1751727607